domainservice/src/main/kotlin/com/makarytskyi/rentcar/service/impl/OrderServiceImpl.kt

```diff
@@ -88,6 +88,12 @@ internal class OrderServiceImpl(
             .flatMap { orderRepository.patch(id, it.toPatch()) }
             .flatMap { order -> getCarPrice(order.carId.toString()).map { order.toResponse(it) } }
 
+
+    override fun findOrderByDateAndCar(date: Date, carId: String): Mono<OrderResponseDto> =
+        orderRepository.findOrderByDateAndCarId(date, carId)
+            .flatMap { it.toMono().zipWith(getCarPrice(it.carId.toString())) }
```

Comment:
Can we simplify this by zipping from scratch?

```suggestion
        Mono.zip(
            orderRepository.findOrderByDateAndCarId(date, carId),
            getCarPrice(carId),
        )
            .map { /* to respose */ }
```